### PR TITLE
[xds test] Add ReturnAll method to a HookServer

### PR DESCRIPTION
**Pull request by eugeneo**
I propose to add a ReturnAll method to a Hook server. This replaces the HookRequest with the Return method. Ultimately, this would allow us to completely remove the SendHookRequest method from the XdsUpdateHealthService when the tests no longer rely on it.

### CODE REVIEW

**Review by gnossen — APPROVED** [review]
Thanks for the change Eugene! Please get Sanjay's approval as well.

File: grpc/testing/test.proto [hunk 1]
```diff
@@ -95,6 +95,7 @@ service LoadBalancerStatsService {
 // with HookRequestCommand::START
 service HookService {
   rpc Hook(grpc.testing.Empty) returns (grpc.testing.Empty);
+  rpc ReturnAll(ReturnAllRequest) returns (grpc.testing.Empty);
```
Comment (sanjaypujare) [inline grpc/testing/test.proto:98]: Why is this needed? This is different from what was in the design.

File: grpc/testing/messages.proto [hunk 2]
```diff
@@ -319,6 +319,14 @@ message TestOrcaReport {
   map<string, double> utilization = 4;
 }
 
+// Request to a ReturnAll method of the HookService. Specifies the status
+// that will be returned to all pending hook requests as well as hook requests
+// that will arrive after this call.
+message ReturnAllRequest {
```
Comment (sanjaypujare) [inline grpc/testing/messages.proto:325]: This message does not seem to serve any purpose

File: grpc/testing/test.proto (hunk 1 above)
Comment (eugeneo) [reply grpc/testing/test.proto:98]: I feel like it is more logical. Current plan is to move the hook service to the maintenance port and have it run all the time. This makes START and STOP hook requests obsolete, leaving only the Return one. I really don't see a need to couple XdsHealthService to the HookService just to forward the Return command.

Comment (sanjaypujare) [reply grpc/testing/test.proto:98]: Okay, discussed offline. The Hook command function and the actual Hook function are combined into a single service. Sounds good.

File: grpc/testing/messages.proto (hunk 2 above)
Comment (eugeneo) [reply grpc/testing/messages.proto:325]: It is the status that will be returned to hook requests, same as https://github.com/grpc/grpc-proto/blob/master/grpc/testing/messages.proto#L334

File: grpc/testing/test.proto (hunk 1 above)
Comment (sanjaypujare) [reply grpc/testing/test.proto:98]: But what about the `reset` functionality described in the doc? Since `ReturnAll` also applies to future `Hook` calls if there is a need to reset this state we should have it, no?

File: grpc/testing/test.proto [hunk 3]
```diff
@@ -95,6 +95,7 @@ service LoadBalancerStatsService {
 // with HookRequestCommand::START
```
Comment (sanjaypujare) [inline grpc/testing/test.proto:95]: Since the HookService serves both the `Hook` calls and the command call (like `ReturnAll`) there is no connection with the HealthService any more. You should update the comment on line 94

**Review by sanjaypujare — CHANGES_REQUESTED** [review]
Pls add the `reset` call too

File: grpc/testing/test.proto (hunk 3 above)
Comment (eugeneo) [reply grpc/testing/test.proto]: Done

File: grpc/testing/test.proto (hunk 1 above)
Comment (eugeneo) [reply grpc/testing/test.proto]: Added ClearReturnStatus

**Review by eugeneo — COMMENTED** [review]
@sanjaypujare I added ClearReturnStatus, renamed setter and added some comments. PTAL.

File: grpc/testing/test.proto [hunk 4]
```diff
@@ -91,10 +91,16 @@ service LoadBalancerStatsService {
       returns (LoadBalancerAccumulatedStatsResponse) {}
 }
 
-// Hook service that may be started on request by calling XdsUpdateHealthService
-// with HookRequestCommand::START
+// Hook service. Used to keep a connection alive so Kubernates does not shut
```
Comment (sanjaypujare) [inline grpc/testing/test.proto:94]: nit: "kubernetes"

Also nit: there is no connection to keep alive. In the current case it is used to keep the pod alive for currently assigned sessions to the pod.

**Review by sanjaypujare — APPROVED** [review]
2 nits found otherwise looks good

File: grpc/testing/test.proto (hunk 4 above)
Comment (eugeneo) [reply grpc/testing/test.proto]: Addressed. Thank you for the review!
